### PR TITLE
WIP: adding update-relationship tests

### PR DESCRIPTION
**Pull request by bobholt**

### CODE REVIEW

File: test/integration/json-api/v1/base-format/update/index.js [hunk 1]
```diff
@@ -270,6 +269,8 @@ describe('updatingResources', function() {
     describe('200Ok', function() {
 
       // TODO: No idea why this suddenly started returning 204 instead of 200
```
Comment (tkellen) [inline test/integration/json-api/v1/base-format/update/index.js]: do these notes need to remain?

File: test/integration/json-api/v1/base-format/update/index.js [hunk 2]
```diff
@@ -398,12 +401,30 @@ describe('updatingRelationships', function() {
...
+      return Agent.request('PATCH', '/books/1/links/series')
+        .send({ data: { what: 'a bad request'}})
+        .promise()
+        .then(function(res) {
+          expect(res.status).to.equal(400);
+        });
+    });
   });
 
   // TODO: Both of these tests seem to be broken for some reason now...
```
Comment (tkellen) [inline test/integration/json-api/v1/base-format/update/index.js]: is this still relevant?

**Comment by tkellen** [comment]
other than the slightly weird commit history, this seems great

File: test/integration/json-api/v1/base-format/update/index.js [hunk 3]
```diff
@@ -439,17 +460,142 @@ describe('updatingRelationships', function() {
...
+      ];
+      return Agent.request('DELETE', '/books/1/links/stores')
+        .send({ data: newIds})
+        .promise()
+        .then(function(res) {
+          expect(res.status).to.equal(204);
+          return Agent.request('GET', '/books/1?include=stores').promise();
+        })
+        .then(function(res) {
+          console.log(res.body.included);
```
Comment (tkellen) [inline test/integration/json-api/v1/base-format/update/index.js]: should be removed

File: test/integration/json-api/v1/base-format/update/index.js (hunk 1 above)
Comment (bobholt) [inline test/integration/json-api/v1/base-format/update/index.js]: I'd like to keep this one for now. Even if json-api relaxes the requirements, they're still RFC 7231 requirements, and I'd like to test for them. The 200 and 409 tests passed at some point.

File: test/integration/json-api/v1/base-format/update/index.js (hunk 2 above)
Comment (bobholt) [inline test/integration/json-api/v1/base-format/update/index.js]: I'll pull this one.

**Comment by bobholt** [comment]
Yeah, I pulled master after your last fix instead of rebasing, so it added the merge commit.

**Pull request merged**